components/brave_wallet/browser/eth_nonce_tracker.h

```diff
@@ -6,46 +6,43 @@
 #ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ETH_NONCE_TRACKER_H_
 #define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ETH_NONCE_TRACKER_H_
 
+#include <memory>
 #include <string>
+#include <vector>
 
-#include "base/callback.h"
-#include "base/memory/raw_ptr.h"
 #include "base/memory/weak_ptr.h"
-#include "base/synchronization/lock.h"
+#include "brave/components/brave_wallet/browser/nonce_tracker.h"
 #include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
 #include "brave/components/brave_wallet/common/brave_wallet_types.h"
 
 namespace brave_wallet {
 
-class EthAddress;
 class JsonRpcService;
-class EthTxStateManager;
+class TxStateManager;
+class TxMeta;
 
-class EthNonceTracker {
+class EthNonceTracker : public NonceTracker {
  public:
-  EthNonceTracker(EthTxStateManager* tx_state_manager,
+  EthNonceTracker(TxStateManager* tx_state_manager,
                   JsonRpcService* json_rpc_service);
-  ~EthNonceTracker();
-  EthNonceTracker(const EthNonceTracker&) = delete;
+  ~EthNonceTracker() override;
   EthNonceTracker operator=(const EthNonceTracker&) = delete;
 
```

Comment:
please help delete copy constructor too.

Reply:
done

Reply:
I meant we should have both copy constructor and copy assignment = delete.

Reply:
done